packages/sign_in_with_apple/lib/src/widgets/sign_in_with_apple_button.dart

```diff
@@ -20,79 +23,157 @@ class SignInWithAppleButton extends StatelessWidget {
+  ///
+  /// Defaults to `8` pixels.
   final BorderRadius borderRadius;
 
+  /// How the icon should be aligned inside the button
+  ///
+  /// Defaults to [IconAlignment.center]
   final IconAlignment iconAlignment;
 
+  /// Returns the background color of the button based on the [style]
+  Color get _backgroundColor {
+    switch (style) {
+      case SignInWithAppleButtonStyle.black:
+        return Colors.black;
+      case SignInWithAppleButtonStyle.white:
+      case SignInWithAppleButtonStyle.whiteOutlined:
+        return Colors.white;
+    }
+
+    assert(false, 'Unknown style: $style');
+    return null;
+  }
+
+  /// Returns the contrast color to the [_backgroundColor]
+  ///
+  /// This is used for the text color and the icon color of the button
+  Color get _contrastColor {
+    switch (style) {
+      case SignInWithAppleButtonStyle.black:
+        return Colors.white;
+      case SignInWithAppleButtonStyle.white:
+      case SignInWithAppleButtonStyle.whiteOutlined:
+        return Colors.black;
+    }
+
+    assert(false, 'Unknown style: $style');
+    return null;
+  }
+
+  /// The decoration which should be applied to the inner container inside the button
+  ///
+  /// This allows to customize the border of the button
+  Decoration get _decoration {
+    switch (style) {
+      case SignInWithAppleButtonStyle.black:
+      case SignInWithAppleButtonStyle.white:
+        return null;
+
+      case SignInWithAppleButtonStyle.whiteOutlined:
+        return BoxDecoration(
+          border: Border.all(width: 1, color: _contrastColor),
+          borderRadius: borderRadius,
+        );
+    }
+
+    assert(false, 'Unknown style: $style');
+    return null;
+  }
+
   @override
   Widget build(BuildContext context) {
-    final backgroundColor =
-        style == SignInWithAppleButtonStyle.dark ? Colors.black : Colors.white;
-    final contrastColor =
-        style == SignInWithAppleButtonStyle.dark ? Colors.white : Colors.black;
+    // per Apple's guidelines
+    final fontSize = height * 0.43;
+
+    final textWidget = Text(
+      text,
+      textAlign: TextAlign.center,
+      style: TextStyle(
+        fontSize: fontSize,
+        color: _contrastColor,
+        // defaults styles aligned with https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/cupertino/text_theme.dart#L16
+        fontFamily: '.SF Pro Text',
+        letterSpacing: -0.41,
+      ),
+    );
+
+    final appleIcon = Container(
+      width: _appleIconSize,
+      height: _appleIconSize,
+      child: Center(
+        child: Container(
+          width: fontSize * (25 / 31),
+          height: fontSize,
+          child: CustomPaint(
+            painter: AppleLogoPainter(
+              color: _contrastColor,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    var children = <Widget>[];
+
+    switch (iconAlignment) {
+      case IconAlignment.center:
+        children = [
+          appleIcon,
+          Flexible(
```

Comment:
Previously the `Flexible` and `Expanded` widgets were missing which could result in overflows

Reply:
Well, I think Apple wouldn't accept if it's being cut off anyway, but we can support that, in case anyone uses dynamic translations and narrow containers.